src/ripple/app/ledger/tests/Ledger_test.cpp

```diff
@@ -0,0 +1,149 @@
+//------------------------------------------------------------------------------
+/*
+    This file is part of rippled: https://github.com/ripple/rippled
+    Copyright (c) 2012, 2013 Ripple Labs Inc.
+
+    Permission to use, copy, modify, and/or distribute this software for any
+    purpose  with  or without fee is hereby granted, provided that the above
+    copyright notice and this permission notice appear in all copies.
+
+    THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+    ANY  SPECIAL ,  DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#include <ripple/app/tests/common_ledger.h>
+
+namespace ripple {
+namespace test {
+
+class Ledger_test : public beast::unit_test::suite
+{
+    void test_genesisLedger (bool sign)
+    {
+        std::uint64_t const xrp = std::mega::num;
+
+        auto master = createAccount ("masterpassphrase");
+
+        Ledger::pointer LCL = createGenesisLedger(100000*xrp, master);
+
+        Ledger::pointer ledger = std::make_shared<Ledger>(false, *LCL);
+
+        // User accounts
+        auto gw1 = createAccount ("gw1");
+        expect (gw1.pk != master.pk, "gw1.pk != master.pk");
```

Comment:
I'll toss in that, at least in my case recently, I added more debug text to some of my unit tests because it was hard to figure out which ones were failing.  They were testing business logic that I was changing.  Text identifying the failing test case is not a waste of time.